sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -1426,33 +1419,30 @@ std::unique_ptr<CurlNetworkConnection> CurlConnectionPool::ExtractOrCreateCurlCo
   // curl-transport adapter supports only HTTP/1.1
   // https://github.com/Azure/azure-sdk-for-cpp/issues/2848
   // The libcurl uses HTTP/2 by default, if it can be negotiated with a server on handshake.
-  if (!SetLibcurlOption(newHandle, CURLOPT_HTTP_VERSION, CURL_HTTP_VERSION_1_1, &result))
+  if (!SetLibcurlOption(newHandle.get(), CURLOPT_HTTP_VERSION, CURL_HTTP_VERSION_1_1, &result))
   {
-    curl_easy_cleanup(newHandle);
     throw Azure::Core::Http::TransportException(
         _detail::DefaultFailedToGetNewConnectionTemplate + hostDisplayName
         + ". Failed to set libcurl HTTP/1.1" + ". " + std::string(curl_easy_strerror(result)));
   }
 
   // Make libcurl to support only TLS v1.2 or later
-  if (!SetLibcurlOption(newHandle, CURLOPT_SSLVERSION, CURL_SSLVERSION_TLSv1_2, &result))
+  if (!SetLibcurlOption(newHandle.get(), CURLOPT_SSLVERSION, CURL_SSLVERSION_TLSv1_2, &result))
   {
-    curl_easy_cleanup(newHandle);
     throw Azure::Core::Http::TransportException(
         _detail::DefaultFailedToGetNewConnectionTemplate + hostDisplayName
         + ". Failed enforcing TLS v1.2 or greater. " + std::string(curl_easy_strerror(result)));
   }
 
-  auto performResult = curl_easy_perform(newHandle);
+  auto performResult = curl_easy_perform(newHandle.get());
   if (performResult != CURLE_OK)
   {
-    curl_easy_cleanup(newHandle);
     throw Http::TransportException(
         _detail::DefaultFailedToGetNewConnectionTemplate + hostDisplayName + ". "
         + std::string(curl_easy_strerror(performResult)));
   }
 
-  return std::make_unique<CurlConnection>(newHandle, connectionKey);
```

Comment:
This will still leak if `std::make_unique<CurlConnection>` throws an exception.

Reply:
weeeeell if i have everytihing oin order and the constructor fails i suspect we would have bigger problems.